src/main/java/net/shadowfacts/shadowlib/version/VersionMatcher.java

```diff
@@ -0,0 +1,146 @@
+package net.shadowfacts.shadowlib.version;
+
+
+
+/**
+ * Used to match version matching strings (e.g. "1.2.x") with versions.
+ * <b>Example:</b>
+ * <code>
+ *     VersionMatcher.matches("1.2.x", new Version("1.2.3"));
+ * </code>
+ * returns true whereas
+ * <code>
+ *     VersionMatcher.matches("1.2.x", new Version("1.3.4"));
+ * </code>
+ * returns false
+ * @author shadowfacts
+ */
+public class VersionMatcher {
+
+	private static final String[] WILDCARD_STRINGS = new String[] {
+			"*", "x", "X"
+	};
+
+	private String matcherString;
+
+	/**
+	 * Constructor used to create a new reusable {@link VersionMatcher}
+	 * @param matcherString
+	 */
+	public VersionMatcher(String matcherString) {
+		this.matcherString = matcherString;
+	}
+
+	/**
+	 * Checks if the {@link Version} is valid for the matcher {@link String} passed to the constructor
+	 * @param v
+	 * @return
+	 */
+	public boolean matches(Version v) {
+		return matches(matcherString, v);
+	}
+
+	/**
+	 * Check if the {@link Version} is valid for the matcher {@link String}
+	 * @param matcherString
+	 * @param v
+	 * @return
+	 */
+	public static boolean matches(String matcherString, Version v) {
+		for (String s : WILDCARD_STRINGS) {
+			if (matcherString.equals(s)) {
+				return true;
+			}
+		}
+
+		if (matcherString.startsWith(">=")) {
+			return  matchesGreaterThanOrEqualTo(matcherString.substring(2), v);
+		} else if (matcherString.startsWith("<=")) {
+			return matchesLessThanOrEqualTo(matcherString.substring(2), v);
+		} else if (matcherString.startsWith("^") || matcherString.startsWith(">")) {
+			return matchesGreaterThan(matcherString.substring(1), v);
+		} else if (matcherString.startsWith("<")) {
+			return matchesLessThan(matcherString.substring(1), v);
+		}
```

Comment:
We need something like "above 1.2.0, but lower than 2.0.0" as well, as MINOR versions can have non-breaking changes.